html/assets/show/machinery.js

```diff
@@ -78,6 +78,7 @@ $(document).ready(function () {
 
   // Set up file download links
   $(".file-download").click(function() {
+    $(".spinner").show();
```

Comment:
I'd make the css selector more specific in order to ensure that you'll not conflict with other spinners in the future.